client/app/bundles/course/assessment/pages/AssessmentShow/AssessmentShowHeader.tsx

```diff
@@ -100,27 +125,29 @@ const AssessmentShowHeader = (
 
       {assessment.submissionsUrl && (
         <Tooltip disableInteractive title={t(translations.submissions)}>
-          <IconButton
-            aria-label={t(translations.submissions)}
-            className="text-white"
-            // TODO: Change to react-router Link once SPA
-            href={assessment.submissionsUrl}
-          >
-            <Inventory />
-          </IconButton>
+          <Link to={assessment.submissionsUrl}>
+            <IconButton
+              aria-label={t(translations.submissions)}
+              className="text-white"
+            >
+              <Inventory />
+            </IconButton>
+          </Link>
         </Tooltip>
       )}
 
-      {assessment.actionButtonUrl && (
-        <Button
-          aria-label={t(ACTION_LABELS[assessment.status])}
-          className="ml-4 bg-white"
-          // TODO: Change to react-router Link once SPA
-          href={assessment.actionButtonUrl}
-          variant="outlined"
-        >
-          {t(ACTION_LABELS[assessment.status])}
-        </Button>
+      {actionButtonUrl && (
+        <Link to={actionButtonUrl}>
+          <Button
+            aria-label={t(ACTION_LABELS[assessment.status])}
+            className="ml-4 bg-white"
+            disabled={attempting}
+            onClick={handleActionButton}
```

Comment:
Yes we did redirection, but it was working only when the backend redirects our GET `/attempt` request. I may be wrong but In SPA, we no longer be able to redirect from the backend. Instead, the frontend needs to receive the redirectUrl and the frontend does the redirection. 

Basically the button acts as as a link, except when attempting a new submission. User can right click the button if its not attempt and it will be opened in new tab. If it is a new attempt, the handleActionButton will prevent the link to be propagated and instead, wait for backend to receive the redirectUrl.

Reply:
So, it means we can no longer attempt in new tab?

This might be an issue, but I get where you're coming from, with Rails' router no longer existing and all. Maybe we will have to handle this in React Router later.

Reply:
Yeah we cant..